### PR TITLE
Code fixes

### PR DESCRIPTION
**Pull request by sbomer**
This applies suggested code fixes except for those (slices) that aren't supported on full framework. I decided not to do the testcases for now, since some of them are arguably more readable without the fixes - and also partially because the omnisharp analyzer only kicks in for open files, and there are a lot of testcases.

### CODE REVIEW

File: src/ILLink.Tasks/FindNativeDeps.cs [hunk 1]
```diff
@@ -53,26 +53,25 @@ public override bool Execute ()
 
 			var managedAssemblies = ManagedAssemblyPaths.Select (i => i.ItemSpec).ToArray ();
 			foreach (string managedAssembly in managedAssemblies) {
-				using (var peReader = new PEReader(new FileStream (managedAssembly, FileMode.Open, FileAccess.Read, FileShare.Read))) {
-					if (peReader.HasMetadata) {
-						var reader = peReader.GetMetadataReader ();
-						for (int i = 1, count = reader.GetTableRowCount (TableIndex.ModuleRef); i <= count; i++) {
-							var moduleRef = reader.GetModuleReference (MetadataTokens.ModuleReferenceHandle (i));
-							var moduleName = reader.GetString (moduleRef.Name);
+				using var peReader = new PEReader(new FileStream (managedAssembly, FileMode.Open, FileAccess.Read, FileShare.Read));
```
Comment (marek-safar) [inline src/ILLink.Tasks/FindNativeDeps.cs]: I am really not sure this using syntax is better. It really hides the scope where the file can be released.

File: src/analyzer/common/Mono.Options/Options.cs [hunk 2]
```diff
@@ -212,29 +212,28 @@ private static IEnumerable<string> CreateWrappedLinesIterator (string self, IEnu
 				yield return string.Empty;
```
Comment (marek-safar) [inline src/analyzer/common/Mono.Options/Options.cs]: Please undo changes in this file, it's a copy of a shared file

File: src/linker/Linker.Steps/BlacklistStep.cs [hunk 3]
```diff
@@ -126,16 +126,14 @@ static ResolveFromXmlStep GetResolveStep (string descriptor)
 
 		protected static XPathDocument GetExternalDescriptor (EmbeddedResource resource)
 		{
-			using (var sr = new StreamReader (resource.GetResourceStream ())) {
-				return new XPathDocument (sr);
-			}
+			using var sr = new StreamReader (resource.GetResourceStream ());
```
Comment (marek-safar) [inline src/linker/Linker.Steps/BlacklistStep.cs]: Is this syntax really better?

Comment (mrvoorhe) [reply src/linker/Linker.Steps/BlacklistStep.cs]: I agree.  I'm not a big fan of the bracketless using

**Comment by sbomer** [comment]
I undid the Options.cs and using changes. PTAL.

**Pull request merged**